gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java

```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
       columnListInQuery = Utils.getColumnListFromQuery(inputQuery);
     }
 
+	// Deepak added the below code for building the exclusion list
+
+	String excludedColumns = workUnit.getProp(ConfigurationKeys.SOURCE_QUERYBASED_EXCLUDED_COLUMNS);
+    List<String> columnListExcluded = new ArrayList<String>();
+    
+    if (Strings.isNullOrEmpty(inputQuery)  && !Strings.isNullOrEmpty(excludedColumns)) {
```

Comment:
Can you add some explanation here, why do you need to check `Strings.isNullOrEmpty(inputQuery)`